polyfill/spatial-navigation-polyfill.js

```diff
@@ -142,19 +142,24 @@
 
     // 1
     const searchOrigin = findSearchOrigin();
-    let eventTarget = null;
+    let eventTarget = searchOrigin;
+
     let elementFromPosition = null;
 
     // 2 Optional step, UA defined starting point
     if (startingPoint) {
-      elementFromPosition = (document.elementFromPoint(startingPoint.x, startingPoint.y)).getSpatialNavigationContainer();
-    }
+      // if there is a starting point, set eventTarget as the element from position for getting the spatnav container
+      elementFromPosition = document.elementFromPoint(startingPoint.x, startingPoint.y);
+
+       // 3 use starting point
+      // 1) starting point is inside the spatnav container
+      // 2) starting point is inside the non-focusable element
 
-    // 3
-    if (elementFromPosition && searchOrigin.contains(elementFromPosition)) {
-      eventTarget = elementFromPosition;
-    } else {
-      eventTarget = searchOrigin;
+      if (elementFromPosition && (searchOrigin === elementFromPosition) && !isContainer(searchOrigin)) {
```

Comment:
Any reason for adding First condition?
Is there any case the `elementFromPosition` become null or undefined??

Reply:
Ah, yes. Good point. 
The first condition, `elementFromPosition` isn't needed because this if-clause is already inside the condition of `if(startingPoint)`.